Common/src/gRPC/ExpressionBuilders.cs

```diff
@@ -0,0 +1,168 @@
+  /// <param name="selector"><see cref="Expression" /> to access the field</param>
+  /// <param name="value">Value for the operation</param>
+  /// <param name="binaryExpression">Operation to perform</param>
+  /// <returns>
+  ///   The <see cref="Expression" /> that represents the operation on the field with the given value
+  /// </returns>
+  public static Expression<Func<TIn, bool>> MakeBinary<TIn, T>(Expression<Func<TIn, object?>> selector,
+                                                               T                              value,
+                                                               ExpressionType                 binaryExpression)
+  {
+    var constant = Expression.Constant(value,
+                                       typeof(T));
+
+    var member = GetMemberExpression(selector);
+
+    var binary = Expression.MakeBinary(binaryExpression,
+                                       member,
+                                       member.Type == typeof(T)
+                                         ? constant
+                                         : Expression.Convert(constant,
+                                                              member.Type));
+
+    return Expression.Lambda<Func<TIn, bool>>(binary,
+                                              selector.Parameters);
+  }
+
+  /// <summary>
+  ///   Generate <see cref="Expression" /> for binary operations
+  /// </summary>
+  /// <typeparam name="TIn">Class from which to access the field</typeparam>
+  /// <typeparam name="T">Type of the value and field on which the operation is applied</typeparam>
+  /// <param name="selector"><see cref="Expression" /> to access the field</param>
+  /// <param name="value">Value for the operation</param>
+  /// <param name="type">Type of the collection containing the method we want to apply</param>
+  /// <param name="method">The name of the method from the type</param>
+  /// <param name="invert">Whether we should invert the result (not operation)</param>
+  /// <returns>
+  ///   The <see cref="Expression" /> that represents the operation on the field with the given value
+  /// </returns>
+  public static Expression<Func<TIn, bool>> MakeCall<TIn, T>(Expression<Func<TIn, object?>> selector,
+                                                             T                              value,
+                                                             Type                           type,
+                                                             string                         method,
+                                                             bool                           invert = false)
+  {
+    var member = GetMemberExpression(selector);
+
+    var constant = Expression.Constant(value,
+                                       typeof(T));
+
+    var methodInfo = type.GetMethods(BindingFlags.Static | BindingFlags.Public)
```

Comment:
Why do you make sure here that the method we get is static and public but we don't do it in the case of string type?

Reply:
Public and static methods are usually extensions methods that will take two parameters : the collection and the value.
For strings, the methods we want to use are not extensions methods and take only one paramter.
By the way, the methods are not called the same way in expressions.